lib/Majkl578/NetteAddons/Doctrine2Identity/DI/IdentityExtension.php

```diff
@@ -25,11 +25,11 @@ class IdentityExtension extends CompilerExtension
 {
 	const NAME = 'doctrine2identity';
 
-	public function loadConfiguration()
+	public function beforeCompile()
 	{
 		$builder = $this->getContainerBuilder();
-
-		$builder->getDefinition('nette.userStorage')
+		$builder->removeDefinition($builder->getByType('Nette\Security\IUserStorage') ?: 'nette.userStorage');
```

Comment:
@fprochazka I have tried it again without removing the definition. Rather I used **getDefinition()->setClass()** but I got exception: *Unable to create service 'security.userStorage', value returned by factory is not Majkl578\NetteAddons\Doctrine2Identity\Http\UserStorage type.* **again**.